dali/core/dummy.cc

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+namespace dali {
+namespace {
+
+// add this alignment to work around a patchelf bug/feature which
+// changes TLS alignment and break DALI interoperability with CUDA RT
+alignas(0x1000) thread_local volatile bool dummy_alignas;
+
+}
```

Comment:
Doesn't putting it in anonymous namespace cause the linker to remove this variable? It's not referenced and has no external linkage, so the compiler might still remove it despite being marked as `volatile`.

Reply:
__attribute__((unused)) maybe?

Reply:
``unused
This attribute, attached to a variable, means that the variable is meant to be possibly unused. GCC will not produce a warning for this variable.```
I doesn't guarantee not to remove it, it will just suppress the warning.

Reply:
```__attribute__((used))``` apparently exists as well xD
https://stackoverflow.com/a/29545417